design.md

```diff
@@ -0,0 +1,84 @@
+# Bazel Module (bzlmod) Support in Renovate
+
+```yaml
+author: Chuck Grindel<chuck.grindel@gmail.com>
+status: To be reviewed
+created: 2023-04-10
+```
+
+## Abstract
+
+[Renovate] is a popular service for updating external dependencies. It [currently supports] updating
+Bazel dependencies described in [WORKSPACE-managed repositories]. With the introduction of [Bazel
+modules], [Renovate] needs to be modified to support the new external dependency mechanism.
+
+## Requirements
+
+- External Dependency Discovery
+  - Support extracting external dependency information from `MODULE.bazel` files.
+    - Read external dependency information from [bazel_dep] declarations.
+    - Support module overrides.
+      - [Single version override]
+      - [Multiple version override]
+      - [Non-registry overrides]
+- Registry Discovery
+  - Support parsing of `.bazelrc` and files imported in the `.bazelrc` looking for [--registry]
```

Comment:
What is the priority for things like this? If we only support bcr to start can we avoid .bazelrc parsing entirely for an initial release?

Reply:
When I create a backlog of work items, this will be lower on the list. I feel the same about the `.bazelversion` and `bazel_compatibility` functionality.